### ISSUE TITLE
Headers Setter Extension Filtering Functionality

### ISSUE THREAD

**Issue by kyo-ke** [open-telemetry/opentelemetry-collector-contrib#35276]
### Component(s)

extension/headerssetter

### Is your feature request related to a problem? Please describe.

if new tenantID come to Tempo, it will automatically create tenant for that ID. 
We want to allow only specific format for tenantID and avoid creating random tenant.

### Describe the solution you'd like

Cerate filtering feature using regular expression on headers setter extension.
only matching tenantID is pushed to Tempo.

**Comment by atoulme** [comment]
Is there a way around this by using the transform processor to drop data if the tenant is not matching? @jpkrohling please review.

**Comment by kyo-ke** [comment]
In my understanding transform processor is for modifying telemetry and tenantID is in header so  we cannot use it for this scenario

**Comment by atoulme** [comment]
Yes, you're correct. I don't know how to manipulate context headers.

**Comment by kyo-ke** [comment]
is it OK to create PR for this feature?

**Comment by jpkrohling** [comment]
This would likely be a good candidate for our middleware extensions: https://github.com/open-telemetry/opentelemetry-collector/issues/7441
